Document/0x04f-Testing-Network-Communication.md

```diff
@@ -431,77 +485,73 @@ $ iptables -t nat -A POSTROUTING -j MASQUERADE
 
 Now you can connect your mobile devices to the access point.
 
-### Network Analyzer Tool
-
-Install a tool that allows you to monitor and analyze the network traffic that will be redirected to your host computer. The two most common network monitoring (or capturing) tools are:
+## MASTG-TECH: Intercepting Xamarin Traffic
 
-- [Wireshark](https://www.wireshark.org "Wireshark") (CLI pendant: [TShark](https://www.wireshark.org/docs/man-pages/tshark.html "TShark"))
-- [tcpdump](https://www.tcpdump.org/tcpdump_man.html "tcpdump")
+Xamarin is a mobile app development platform that allows developers to create [native Android](https://docs.microsoft.com/en-us/xamarin/android/get-started/ "Getting Started with Android") and [iOS apps](https://docs.microsoft.com/en-us/xamarin/ios/get-started/ "Getting Started with iOS") using Visual Studio and C#.
 
-Wireshark offers a GUI and is more straightforward if you are not used to the command line. If you are looking for a command line tool you should either use TShark or tcpdump. All of these tools are available for all major Linux and Unix operating systems and should be part of their respective package installation mechanisms.
+When testing a Xamarin app, setting the system proxy in the Device Wi-Fi settings will not capture any HTTP requests in your interception proxy. This is because Xamarin apps do not use the local proxy settings of your device. There are three ways to bypass this limitation:
 
-### Setting a Proxy Through Runtime Instrumentation
+### Option 1: Manipulating Xamarin's Network Stack Default Proxy
```

Comment:
Some thoughts:
- Do we really want to have separate sections for cross-platform frameworks? Generally we tried to stay away from that.
- Do we want to modify source code? (vs dynamic patching)

Reply:
The content for patching was already there but we can remove it and leave the new content that uses Frida.